### DISCUSSION TITLE
可以在非react上下文里进行jotai的get和set操作吗？

### DISCUSSION THREAD

**Discussion by my3188** [pmndrs/jotai#2208] [Q&A]
大多数场景下，都是在react上下文中使用jotai,这是没有问题的，但是仍然在少数场景下，需要在非react上下文设置获取读取jotai中的状态值

**Comment by yf-yang** [comment]
Are you still using React as the framework or you are using other frameworks like vue, etc.?

If you are using React, then could you please provide a general example? I think you can always connect React world with other stuff via `useEffect` or other hooks. So you still get the value by calling `useAtomValue` during the main renderer, then pass it to other stuff in an async effect callback.

If you are using a completely different framework. For now, jotai is tightly coupled with React. However, you can apply similar mechanisms in other frameworks yourself. Check https://jotai.org/docs/guides/core-internals.

**Comment by himself65** [comment]
use `getDefaultStore().get(aAtom)`

**Reply by my3188** [reply]
getDefaultStore().get(aAtom)我试过了，确实可以，谢谢 👍
getDefaultStore().set(aAtom,val),这个我测试不生效，不知道为什么？https://codesandbox.io/s/jotai-demo-forked-ry3nmy?file=/src/App.js

**Reply by himself65** [reply]
const App = () => (
  <Provider store={getDefaultStore()}>
    <Input />
    <CharCount />
    <Uppercase />
  </Provider>
);

**Reply by my3188** [reply]
对不起，我不是很理解你的回复？是说getDefaultStore().set(aAtom,val)必须在App = ()=>{ }内部？

**Reply by himself65** [reply]
抱歉可能是昨晚粘贴错了，我的意思是你需要在provider里面设置store，不然两者不是一个store会导致数据不一致

<img width="412" alt="image" src="https://github.com/pmndrs/jotai/assets/14026360/23cafed9-e9c4-4697-a2e7-c7075a8ce075">

**Reply by my3188** [reply]
如果这样子，我感觉还是在react上下文里，我的意思就是在App之外。例如我有一个a.js,这个js被谁引用都是有可能，我希望在这个a.js中set|get jotai中的状态

**Reply by himself65** [reply]
事实上，如果你不传store，在react hooks内部会用默认的store（理论上这两个应该是一样的store）

https://github.com/pmndrs/jotai/blob/f5691b5a6a68e5b0f5ae05390a45bff309900ea3/src/react/useAtomValue.ts#L60

**Reply by himself65** [reply]
我的解决办法是，你把store当做一个全局常量

**Reply by my3188** [reply]
现在从测试效果来看，在非react上下文只能get不能set, 另外不知道是否在非react上下文中支持订阅，我挺喜欢jotai，但是我又希望它具备zustand的能力

**Reply by 0x7Jia** [reply]
我试了下你的demo, 把包裹的Provider删掉就能正常工作了
```
const App = () => (
  <>
    <Input />
    <CharCount />
    <Uppercase />
  </>
);
```
